lib/vrl/compiler/src/expression/variable.rs

```diff
@@ -63,6 +59,10 @@ impl Expression for Variable {
             .unwrap_or_else(|| TypeDef::null().infallible())
     }
 
+    fn as_value(&self) -> Option<Value> {
+        self.value.clone()
```

Comment:
I think this will still be broken since if you do:

```coffee
if .thing {
  variable = "zub"
} else {
  variable = "buz"
}

variable
```

The variable will always  `as_value` as "buz" since the value is inserted in the else clause of the if statement overwriting the initial value.

As far as I am aware this hasn't been fixed yet, and I do wonder if we should fix that first before expanding this.

Reply:
I am currently working on a fix for that, so I'm fine leaving this PR open for another couple of days.